src/switch_core_media.c

```diff
@@ -4843,58 +4843,68 @@ static switch_status_t check_ice(switch_media_handle_t *smh, switch_media_type_t
-				if (!strcasecmp(val, "passthru")) {
-					switch_log_printf(SWITCH_CHANNEL_SESSION_LOG(smh->session), SWITCH_LOG_INFO, "Activating %s RTCP PASSTHRU PORT %d\n",
-									  type2str(type), remote_rtcp_port);
-					switch_rtp_activate_rtcp(engine->rtp_session, -1, remote_rtcp_port, engine->rtcp_mux > 0);
-				} else {
-					int interval = atoi(val);
-					if (interval < 100 || interval > 500000) {
-						switch_log_printf(SWITCH_CHANNEL_SESSION_LOG(smh->session), SWITCH_LOG_ERROR,
-										  "Invalid rtcp interval spec [%d] must be between 100 and 500000\n", interval);
-						interval = 5000;
-					}
 
-					switch_log_printf(SWITCH_CHANNEL_SESSION_LOG(smh->session), SWITCH_LOG_INFO, "Activating %s RTCP PORT %d\n", type2str(type), remote_rtcp_port);
-					switch_rtp_activate_rtcp(engine->rtp_session, interval, remote_rtcp_port, engine->rtcp_mux > 0);
+				if (remote_rtcp_port) {
+					if (!strcasecmp(val, "passthru")) {
+						switch_log_printf(SWITCH_CHANNEL_SESSION_LOG(smh->session), SWITCH_LOG_INFO,
+										  "Activating %s RTCP PASSTHRU PORT %d\n",
+										  type2str(type), remote_rtcp_port);
+						switch_rtp_activate_rtcp(engine->rtp_session, -1, remote_rtcp_port, engine->rtcp_mux > 0);
+					} else {
+						int interval = atoi(val);
+						if (interval < 100 || interval > 500000) {
+							switch_log_printf(SWITCH_CHANNEL_SESSION_LOG(smh->session), SWITCH_LOG_ERROR,
+											  "Invalid rtcp interval spec [%d] must be between 100 and 500000\n",
+											  interval);
+							interval = 5000;
+						}
+
+						switch_log_printf(SWITCH_CHANNEL_SESSION_LOG(smh->session), SWITCH_LOG_INFO,
+										  "Activating %s RTCP PORT %d\n", type2str(type), remote_rtcp_port);
+						switch_rtp_activate_rtcp(engine->rtp_session, interval, remote_rtcp_port, engine->rtcp_mux > 0);
+					}
 				}
-			}
-		}
 
-		if (engine->rtp_session && engine->ice_in.cands[engine->ice_in.chosen[1]][1].ready) {
-			if (engine->rtcp_mux > 0 && !strcmp(engine->ice_in.cands[engine->ice_in.chosen[1]][1].con_addr, engine->ice_in.cands[engine->ice_in.chosen[0]][0].con_addr)
-				&& engine->ice_in.cands[engine->ice_in.chosen[1]][1].con_port == engine->ice_in.cands[engine->ice_in.chosen[0]][0].con_port) {
-				switch_log_printf(SWITCH_CHANNEL_SESSION_LOG(smh->session), SWITCH_LOG_INFO, "Skipping %s RTCP ICE (Same as RTP)\n", type2str(type));
-			} else {
-				switch_log_printf(SWITCH_CHANNEL_SESSION_LOG(smh->session), SWITCH_LOG_INFO, "Activating %s RTCP ICE\n", type2str(type));
-
-				switch_rtp_activate_ice(engine->rtp_session,
-										engine->ice_in.ufrag,
-										engine->ice_out.ufrag,
-										engine->ice_out.pwd,
-										engine->ice_in.pwd,
-										IPR_RTCP,
-#ifdef GOOGLE_ICE
-										ICE_GOOGLE_JINGLE,
-										NULL
-#else
-										switch_determine_ice_type(engine, smh->session),
-										&engine->ice_in
-#endif
-										);
-			}
+				if (engine->rtp_session && engine->ice_in.cands[engine->ice_in.chosen[1]][1].ready) {
```

Comment:
This should not be commited. Can you please clean this up @StefanYohansson (maybe you need to checkout the file and re-apply your changes only to relevant part - around lines 4850 - 4880